### PR TITLE
allPlatforms system test for Xcode and Android Studio

### LINKED ISSUE #521: iOS and OS X system test
Continous build both targets, including tests.

### PR DESCRIPTION
**Pull request by brunobowden**
Xcode 'allPlatforms' build:
- TODO: needs setup from run-test.sh
- TODO: get watchOS build working
- iOS and OS X builds from shared project
- App and unit tests for iOS and OS X
- All generated from project defaults with minimal additional setup
- Notable configuration:
    * iOS 7.0 (required for Swift)
    * OS X 10.4 (minimum supported by Xcode 7)
    * watchOS 2.0 (minimum for native watch apps)
    * Bitcode disabled until plugin handle bitcode build

Android 'allPlatforms' build:
- 'shared' project + unit test
- Application depends on shared project

Other Changes:
- .gitignore for Xcode and Android Studio files

Fixes #521

TESTED=Manually
- iOS 8.1 simulator
- iOS 8.1 simulator unit tests
- iOS 9.0 iPhone 6 Plus physical device
- OS X 10.11 Mac App
- OS X 10.11 Mac App unit tests
- Android app simulator
- Android app Nexus 6 physical device

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1 - ignore this for now as it's missing most of the crucial files

**Comment by brunobowden** [comment]
@advayDev1 - PTAL, it's now updated with the correct files. Also, no mention of 'brunobowden' anywhere in the files. No automatic setup yet but manually it works.

File: systemTests/simple1/build.gradle [hunk 1]
```diff
@@ -37,8 +37,12 @@ repositories {
 }
 
 dependencies {
+    compile project(':shared')
+
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile 'com.android.support:appcompat-v7:21.0.3'
```
Comment (advayDev1) [inline systemTests/simple1/build.gradle]: this will not work

File: systemTests/simple1/build.gradle [hunk 2]
```diff
@@ -37,8 +37,12 @@ repositories {
 }
 
 dependencies {
+    compile project(':shared')
```
Comment (advayDev1) [inline systemTests/simple1/build.gradle]: there is no such project.  simple1 is a single project.

File: systemTests/allPlatforms/ios/OSX-APP/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -0,0 +1,681 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (advayDev1) [inline systemTests/allPlatforms/ios/OSX-APP/Base.lproj/Main.storyboard]: Does this storyboard really need to be so large?  Can we pare anything down?

File: systemTests/allPlatforms/app/src/test/java/com/github/j2objc_contrib/android_app/ExampleUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * To work on unit tests, switch the Test Artifact in the Build Variants view.
+ */
+public class ExampleUnitTest {
+    @Test
+    public void addition_isCorrect() throws Exception {
+        assertEquals(4, 2 + 2);
+    }
+}
```
Comment (advayDev1) [inline systemTests/allPlatforms/app/src/test/java/com/github/j2objc_contrib/android_app/ExampleUnitTest.java]: new line

**Comment by advayDev1** [comment]
not much for me to comment on here as it seems 90% auto-generated.  i'd recommend waiting for and merging to #518, as your build will push Travis over the timelimit.

File: systemTests/allPlatforms/ios/OSX-APP/Base.lproj/Main.storyboard (hunk 3 above)
Comment (brunobowden) [inline systemTests/allPlatforms/ios/OSX-APP/Base.lproj/Main.storyboard]: It probably can be paired down but I don't have enough knowledge of Xcode to do that with confidence. Infact, I'd be loathe to try for some unintended consequences that I don't understand.

File: systemTests/allPlatforms/app/src/test/java/com/github/j2objc_contrib/android_app/ExampleUnitTest.java (hunk 4 above)
Comment (brunobowden) [inline systemTests/allPlatforms/app/src/test/java/com/github/j2objc_contrib/android_app/ExampleUnitTest.java]: done

File: systemTests/simple1/build.gradle (hunk 2 above)
Comment (brunobowden) [inline systemTests/simple1/build.gradle]: The changes to simple1/build.gradle were the result of a bad merge and shouldn't have been there. Now reverted.

**Comment by brunobowden** [comment]
Right now it's not building anything. So it won't impact the CI at all. Getting that work will obviously depend on #518.

**Comment by advayDev1** [comment]
Sorry - that was my misunderstanding from the modification to simple1/build.gradle - LGTM then

**Pull request merged**